New translation: layout-components.md (#269)
gatsbyjs/gatsby-pt-BR: docs/docs/layout-components.md

---
title: Componentes de layout
---

> Nesse guia você vai entender qual a abordagem do Gatsby em relação a layouts, como criar, usar componentes de layout e como evitar que esses componentes desmontem.

Reviewer: ```suggestion
Neste guia você vai entender qual a abordagem do Gatsby em relação a layouts, como criar, usar componentes de layout e como evitar que esses componentes desmontem.
```

## Abordagem da Gatsby para layouts

> Por padrão, o gatsby não aplica o layout nas páginas de forma automática(Existem formas de fazer isso, mas serão abordadas em uma seção futura). Ao invés disso, o Gastsby segue o modelo composicional do React na sua forma de importar e usar componentes. Isso faz com que seja possível criar multiplos níveis de layout, como por exemplo, um cabeçalho e rodapé globais e em algumas páginas um menu lateral. Isso faz com que a transição de dados entre o layout e os componentes da página seja mais fácil de ser concretizada.

Reviewer: ```suggestion
Por padrão, o gatsby não aplica o layout nas páginas de forma automática(Existem formas de fazer isso, mas serão abordadas em uma sessão futura). Ao invés disso, o Gastsby segue o modelo composicional do React na sua forma de importar e usar componentes. Isso faz com que seja possível criar múltiplos níveis de layout, como por exemplo, um cabeçalho e rodapé globais e em algumas páginas um menu lateral. Isso faz com que a transição de dados entre o layout e os componentes da página seja mais fácil de ser concretizada.
```

> Por padrão, o gatsby não aplica o layout nas páginas de forma automática(Existem formas de fazer isso, mas serão abordadas em uma sessão futura). Ao invés disso, o Gastsby segue o modelo composicional do React na sua forma de importar e usar componentes. Isso faz com que seja possível criar múltiplos níveis de layout, como por exemplo, um cabeçalho e rodapé globais e em algumas páginas um menu lateral. Isso faz com que a transição de dados entre o layout e os componentes da página seja mais fácil de ser concretizada.

Reviewer 2: ```suggestion
Por padrão, Gatsby não aplica o layout nas páginas de forma automática (Existem formas de fazer isso, mas serão abordadas em uma sessão futura). Ao invés disso, Gastsby segue o modelo composicional do React na sua forma de importar e usar componentes. Isso faz com que seja possível criar múltiplos níveis de layout, como por exemplo, um cabeçalho e rodapé globais e em algumas páginas um menu lateral. Isso faz com que a transição de dados entre o layout e os componentes da página seja mais fácil de ser concretizada.
```

)
```

Repita isso para todas as páginas e templates que precisarem desse layout.

> ## Como previnir que os componentes de layout desmontem 

Reviewer: ```suggestion
## Como prevenir que os componentes de layout desmontem 
```

> Como mencionado anteriormente, por padrão o Gastby não empacota as páginas que estão dentro de um componente de layout de forma automático. O componente "top level" é a própria página. Dessa forma, quando o componente "top level" muda de uma página pra outra o React vai re-renderizar todos os seus filhos. Isso significa que os componentes compartilhados, como navegação por exemplo, irão ser desmontados e remontados. Isso causará a quebra das transições do CSS ou do estado do React dentro daqueles componentes compartilhados.

Reviewer: ```suggestion
Como mencionado anteriormente, por padrão o Gatsby não empacota as páginas que estão dentro de um componente de layout de forma automático. O componente "top level" é a própria página. Dessa forma, quando o componente "top level" muda de uma página para outra o React vai re-renderizar todos os seus filhos. Isso significa que os componentes compartilhados, como navegação por exemplo, irão ser desmontados e remontados. Isso causará a quebra das transições do CSS ou do estado do React dentro daqueles componentes compartilhados.
```

> Se você precisar criar um componente wrapper junto a componentes da página que não serão desmontados com as alterações dessa página, use a [API de navegação](/docs/browser-apis/#wrapPageElement) **`wrapPageElement`** e o [SSR equivalente](/docs/ssr-apis/#wrapPageElement).

Reviewer: ```suggestion
Se você precisar criar um componente wrapper junto a componentes da página que não serão desmontados com as alterações desta página, use a [API de navegação](/docs/browser-apis/#wrapPageElement) **`wrapPageElement`** e o [SSR equivalente](/docs/ssr-apis/#wrapPageElement).
```
